test/support/utils_sycl_defs.h

```diff
@@ -0,0 +1,66 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+// This file contains SYCL specific macros and abstractions
+// to support different versions of SYCL and to simplify its interfaces
+//
+// Functionality, required for the tests, is copied from sycl_defs.h and thus is not directly used
+// for compatibility with different versions of the library
+
+#ifndef _UTILS_SYCL_DEFS_H
+#define _UTILS_SYCL_DEFS_H
```

Comment:
We already have `utils_sycl.h`. Do we really need a new header file?

Reply:
Doing so, we would need to include `utils_sycl.h` in `sycl_alloc_utils.h` to use `TEST_LIBSYCL_VERSION` macro. 
This will create a circular dependency. Currently, `utils_sycl.h` includes `utils_test_base.h`, which includes `sycl_alloc_utils.h`.

However, this can be avoided if use `__LIBSYCL_MAJOR_VERSION` and `__LIBSYCL_MINOR_VERSION` macros directly in `sycl_alloc_utils.h`. Let me know, if you preferer that option.